protocol/str.go

```diff
@@ -0,0 +1,19 @@
+package protocol
+
+import "github.com/coniks-sys/coniks-go/merkletree"
+
+type DirSTR struct {
+	merkletree.SignedTreeRoot
```

Comment:
Yes, they will be gc'd. But I think saving some copies can be huge save when the server needs to handle multiple requests? I'm so nitpicking :(